### PR TITLE
executor: fix unstable test TestPlanCacheWithStaleReadByBinaryProto

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/33349

Problem Summary:

### What is changed and how it works?

TestPlanCacheWithStaleReadByBinaryProto is unstable due to local time skew error, this request fixed it.

I run `go test -run TestPlanCacheWithStaleReadByBinaryProto -count 10` locally, the result is pass as expected.

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/stale_txn_test.go [hunk 1]
```diff
@@ -1302,23 +1302,17 @@ func TestPlanCacheWithStaleReadByBinaryProto(t *testing.T) {
 	tk.MustExec("use test")
 	tk.MustExec("create table t1 (id int primary key, v int)")
 	tk.MustExec("insert into t1 values(1, 10)")
+	time.Sleep(time.Second)
 	se := tk.Session()
-	tk.MustExec("set @a=now(6)")
-	time.Sleep(time.Millisecond * 5)
+	now := time.Now()
+	time.Sleep(time.Second * 2)
```
Comment (JmPotato) [inline executor/stale_txn_test.go:1308]: Maybe 1 second is enough.

Comment (lcwangchao) [inline executor/stale_txn_test.go:1308]: Maybe the following change is enough:

```
se := tk.Session()
time.Sleep(time.Millisecond * 100)
tk.MustExec("set @a=now(6)")
tk.MustExec("update t1 set v=100 where id=1")
```

See: https://github.com/pingcap/tidb/pull/32959/files#diff-6f8c739b7faa2c14e5a2e28f2a18abfe12ffe1bf45d4bed9add7dfec69b37710R118-L120 . Only add a sleep before `tk.MustExec("set @a=now(6)")` and 100ms is enough.

Comment (lcwangchao) [inline executor/stale_txn_test.go:1308]: Maybe the following change is enough:

```
se := tk.Session()
time.Sleep(time.Millisecond * 100)
tk.MustExec("set @a=now(6)")
tk.MustExec("update t1 set v=100 where id=1")
```

See: https://github.com/pingcap/tidb/pull/32959/files#diff-6f8c739b7faa2c14e5a2e28f2a18abfe12ffe1bf45d4bed9add7dfec69b37710R118-L120 . Only add a sleep before `tk.MustExec("set @a=now(6)")` and 100ms is enough.

**Review by JmPotato — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**